source/grid/tria_description.cc

```diff
@@ -442,11 +442,11 @@ namespace TriangulationDescription
             tria, coinciding_vertex_groups, vertex_to_coinciding_vertex_group);
         }
 
-        template <typename T>
-        T
+        template <typename DescriptionType>
+        DescriptionType
         create_description_for_rank(const unsigned int my_rank) const
         {
-          T construction_data;
+          DescriptionType construction_data;
```

Comment:
Could you add something like

```cpp
static_assert(std::is_same_v<DescriptionType, Description<dim, spacedim> || std::is_same_v<DescriptionType, DescriptionTemp<dim, spacedim>);
```
here to better document which types this works with?

Reply:
I put that in a follow-up patch that I'll submit if you were willing to merge this as is.

Reply:
Sounds good!